app/api/groups/groups.go

```diff
@@ -44,3 +44,35 @@ func checkThatUserOwnsTheGroup(store *database.DataStore, user *database.User, g
 	}
 	return service.NoError
 }
+
+func checkThatUserHasRightsForDirectRelation(store *database.DataStore, user *database.User, parentGroupID, childGroupID int64) service.APIError {
+	groupStore := store.Groups()
+
+	var groupData []struct {
+		ID   int64  `gorm:"column:ID"`
+		Type string `gorm:"column:sType"`
+	}
+
+	err := groupStore.OwnedBy(user).
+		WithWriteLock().
```

Comment:
don't you have to check that it is run in a transaction?

Reply:
1. WithWriteLock() panics itself when it is called without a transaction.
2. The method is private (unexported) so I don't think we should have a test for this.

Reply:
And we have a test that checks WithWriteLock() panics when outside a transaction